rust/lance-index/src/scalar/inverted/index.rs

```diff
@@ -266,38 +266,33 @@ impl ScalarIndex for InvertedIndex {
 #[derive(Debug, Clone, Default, DeepSizeOf)]
 pub struct TokenSet {
     // token -> (token_id, frequency)
-    tokens: HashMap<String, (u32, u64)>,
+    tokens: HashMap<String, u32>,
     next_id: u32,
 }
 
 impl TokenSet {
     pub fn to_batch(&self) -> Result<RecordBatch> {
         let mut tokens = Vec::with_capacity(self.tokens.len());
         let mut token_ids = Vec::with_capacity(self.tokens.len());
-        let mut frequencies = Vec::with_capacity(self.tokens.len());
-        self.tokens
-            .iter()
-            .for_each(|(token, (token_id, frequency))| {
-                tokens.push(token.clone());
-                token_ids.push(*token_id);
-                frequencies.push(*frequency);
-            });
-        let token_col = StringArray::from(tokens);
-        let token_id_col = UInt32Array::from(token_ids);
-        let frequency_col = UInt64Array::from(frequencies);
+        self.tokens.iter().for_each(|(token, token_id)| {
+            tokens.push(token.clone());
+            token_ids.push(*token_id);
+        });
+        let mut indices = (0..self.tokens.len()).collect_vec();
+        indices.sort_unstable_by_key(|&i| tokens[i].as_str());
```

Comment:
SHould we just use a `tokens: BTreeMap<...>`?

Reply:
that can avoid the sorting, but we need faster `get` to map token string to token ID, so use `HashMap` here